lib/json_schemer.rb

```diff
@@ -37,10 +37,14 @@ class InvalidSymbolKey < StandardError; end
 
   DEFAULT_META_SCHEMA = 'http://json-schema.org/draft-07/schema#'
 
+  WINDOWS_URI_PATH_REGEX = /\A\/[a-z]:/i
+
   FILE_URI_REF_RESOLVER = proc do |uri|
     raise InvalidFileURI, 'must use `file` scheme' unless uri.scheme == 'file'
     raise InvalidFileURI, 'cannot have a host (use `file:///`)' if uri.host && !uri.host.empty?
-    JSON.parse(File.read(uri.path))
+    path = uri.path
+    path = path[1..-1] if path.match?(WINDOWS_URI_PATH_REGEX)
```

Comment:
@emiltin I took your advice and replaced the logic here with a single regex. I was trying to keep it as close as possible to [`correct_for_windows_path`](https://github.com/ruby/ruby/blob/48b94b791997881929c739c64f95ac30f3fd0bb9/lib/rubygems/util.rb#L111), but there's not really much point as long as it's the same logic.